innkeeper/clientauth.go

```diff
@@ -0,0 +1,32 @@
+package innkeeper
+
+import (
+	"github.com/zalando/skipper/oauth"
+)
+
+// An Authentication object provides authentication to Innkeeper.
+type Authentication interface {
```

Comment:
if the innkeeper package will import the oauth package directly, then this public interface doesn't seem to make much sense. Am I wrong? The original goal was to have less dependencies between the packages, but it's fine, but then we don't really need the extra interface.